pkg/build/api/validation/validation.go

```diff
@@ -54,6 +55,29 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) field.Err
 	return allErrs
 }
 
+// ValidateBuildDetailsUpdate validates build details update
+func ValidateBuildDetailsUpdate(build *buildapi.Build, older *buildapi.Build) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if older.Spec.Revision != nil {
+		// If there was already a revision, then return an error
+		allErrs = append(allErrs, field.Duplicate(field.NewPath("status", "revision"), older.Spec.Revision))
+	}
+	if build.Spec.Revision == nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("status", "revision"), nil, "cannot set an empty revision in build status"))
+	}
+	return allErrs
+}
+
+// ValidateBuildStatusUpdate validates a build status update
+func ValidateBuildStatusUpdate(build *buildapi.Build, older *buildapi.Build) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// Ensure that the cancelled flag is not changed in a status update
+	if older.Status.Cancelled != build.Status.Cancelled {
```

Comment:
Cancelled is user input.  It should be reflected into status, but they are two separate fields.

1. Users set spec cancelled.
2. Build controller reacts with status cancelled.

Not the same field.